lib/chat_app/ets_database.ex

```diff
@@ -0,0 +1,12 @@
+defmodule ChatApp.ETSDatabase do
+  @tables [
+    %{name: :rooms, properties: [:set, :public, :named_table]},
+    %{name: :messages, properties: [:set, :public, :named_table]}
+  ]
+
+  def initialize() do
+    Enum.map(@tables, fn row ->
+      {row.name, row.properties}
+    end)
+  end
```

Comment:
Maybe the name `initialize` is not a good fit since it does not initialize anything - it just prepares the data to be used by the `tables` start phase, where the initialization happens.

Reply:
i would suggest defining a behaviour for the Db module that will define the interface we need.

i could contain callbacks like: 

- init() -one time db setup with all tables
- createRoom()
- addToRoom()
- removeFromRoom()
- etc..